lazy_theta_star_p_planner/src/lazy_theta_star_p.cpp

```diff
@@ -0,0 +1,254 @@
+      }
+    }
+  }
+}
+
+void LazyThetaStarP::backtrace(std::vector<coordsW> & raw_points, int curr_id)
+{
+  std::vector<coordsW> path_rev;
+  coordsW world{};
+  double pusher_ = 0.5 * costmap_->getResolution();
+  do {
+    costmap_->mapToWorld(nodes_data[curr_id].x, nodes_data[curr_id].y, world.x, world.y);
+    path_rev.push_back({world.x + pusher_, world.y + pusher_});
+    if (path_rev.size() > 1) {
+      curr_id = nodes_data[curr_id].parent_id;
+    }
+  } while (curr_id != 0);
+  costmap_->mapToWorld(nodes_data[curr_id].x, nodes_data[curr_id].y, world.x, world.y);
+  path_rev.push_back({world.x + pusher_, world.y + pusher_});
+
+  raw_points.reserve(path_rev.size());
+  for (int i = static_cast<int>(path_rev.size()) - 1; i >= 0; i--) {
+    raw_points.push_back(path_rev[i]);
+  }
+}
+
+bool LazyThetaStarP::losCheck(
+  const int & x0,
+  const int & y0,
+  const int & x1,
+  const int & y1,
+  double & sl_cost)
+{
+  sl_cost = 0;
+
+  int cx, cy;
+  int dy = abs(y1 - y0), dx = abs(x1 - x0), f = 0;
+  int sx, sy;
+  int lx, ly;  // this variable is reserved for future use
+  sy = y1 > y0 ? 1 : -1;
+  sx = x1 > x0 ? 1 : -1;
+
+  int u_x = (sx - 1) / 2;
+  int u_y = (sy - 1) / 2;
+  lx = x1;
+  ly = y1;
+  cx = x0;
+  cy = y0;
+
+  if (dx >= dy) {
+    while (cx != lx) {
+      f += dy;
+      if (f >= dx) {
+        if (!isSafe(cx + u_x, cy + u_y, sl_cost)) {
+          return false;
+        }
+        cy += sy;
+        f -= dx;
+      }
+      if (f != 0 && !isSafe(cx + u_x, cy + u_y, sl_cost)) {
+        return false;
+      }
+      if (dy == 0 && !isSafe(cx + u_x, cy, sl_cost) && !isSafe(cx + u_x, cy - 1, sl_cost)) {
+        return false;
+      }
+      cx += sx;
+    }
+  } else {
+    while (cy != ly) {
+      f = f + dx;
+      if (f >= dy) {
+        if (!isSafe(cx + u_x, cy + u_y, sl_cost)) {
+          return false;
+        }
+        cx += sx;
+        f -= dy;
+      }
+      if (f != 0 && !isSafe(cx + u_x, cy + u_y, sl_cost)) {
+        return false;
+      }
+      if (dx == 0 && isSafe(cx, cy + u_y, sl_cost) && !isSafe(cx - 1, cy + u_y, sl_cost)) {
+        return false;
+      }
+      cy += sy;
+    }
+  }
+  return true;
+}
+
+void LazyThetaStarP::setContainers()
+{
+  index_generated = 0;
+  int last_size_x = size_x_;
+  int last_size_y = size_y_;
+  int curr_size_x = static_cast<int>(costmap_->getSizeInCellsX());
+  int curr_size_y = static_cast<int>(costmap_->getSizeInCellsY());
+  if (last_size_x != curr_size_x || last_size_y != curr_size_y) {
+    initializePosn(curr_size_y * curr_size_x - last_size_y * last_size_x);
+    nodes_data.reserve(curr_size_x * curr_size_y);
+  } else {
+    initializePosn();
+  }
+  size_x_ = curr_size_x;
+  size_y_ = curr_size_y;
+}
+
+void LazyThetaStarP::initializePosn(int size_inc)
+{
+  int i = 0;
+
+  if (!node_position.empty()) {
+    for (; i < size_x_ * size_y_; i++) {
+      node_position[i] = -1;
+    }
+  }
+
+  for (; i < size_inc; i++) {
```

Comment:
You want to use the same `i` here as in the other loop? If that was incremented, you wouldn't be starting at `0` here

Reply:
As the space has already been allocated according to the previous map size, I just reset their values to -1 and then increase the size of the vector (nodes_posistion), to account for an increase in map size of any.